File: algorithms/1_easy/S/03687_smallest_absent_positive_greater_than_average/solution.py
class Solution:
    def smallestAbsent(self, nums: List[int]) -> int:
        sum_nums = sum(nums)
        avg = sum_nums // len(nums)

        num_set = set()
        for  num in nums:
            if num > avg and num > 0:
                num_set.add(num)
        res = max(1, avg + 1)
        while res in num_set:
            res += 1

        return res

File: algorithms/1_easy/S/03687_smallest_absent_positive_greater_than_average/solution.cpp
class Solution {
public:
    int smallestAbsent(vector<int>& nums) {
        int avg{std::reduce(nums.begin(), nums.end(), 0) / static_cast<int>(nums.size())};
        unordered_set<int> set{};
    
        for (const int &num : nums) {
            if (num > avg && num > 0) {
                set.insert(num);
            }
        }

        int res = max(1, avg + 1);
        while (set.contains(res)) {
            res += 1;
        }

        return res;
    }
};
